File: e270_transpose_text/python/Transpose.py
import sys

def main(file_name):
    input_file = open(file_name, 'r')

    # read file and determine maxLength
    lines = []
    for line in input_file:
        lines.append(line.rstrip())
    maxLength = len(max(lines, key=len))

    # Transpose
    result = []
    for i in range(0, maxLength):
        transposed = ""
        for line in lines:
            if i < len(line):
                transposed += line[i]
            else:
                transposed += " "
        result.append(transposed)

    # Print Result
    print "\n".join(result)

main(sys.argv[1])

File: e270_transpose_text/js/Transpose.js
const fs = require('fs')
const filename = process.argv[2];

fs.readFile(filename, 'utf8', (err, lines) => {
    lines = lines.split('\n');

    // determine maxLength
    var maxLength = 0;
    lines.forEach(function(line){
        if (line.length > maxLength) maxLength = line.length;
    });

    // Transpose
    var result = [];
    for (let i = 0; i < maxLength; i++) {
        transposed = ""
        lines.forEach(function(line) {
            transposed += (i < line.length) ? line[i] : ' ';
        });
        result[i] = transposed;
    }

    // Print Result
    console.log(result.join('\n'));
});